devito/symbolics/printer.py

```diff
@@ -122,15 +125,22 @@ def _print_math_func(self, expr, nest=False, known=None):
         return f'{cname}({args})'
 
     def _print_Pow(self, expr):
-        # Need to override because of issue #1627
-        # E.g., (Pow(h_x, -1) AND h_x.dtype == np.float32) => 1.0F/h_x
-        try:
-            if expr.exp == -1 and self.single_prec():
-                PREC = precedence(expr)
-                return '1.0F/%s' % self.parenthesize(expr.base, PREC)
-        except AttributeError:
-            pass
-        return super()._print_Pow(expr)
+        # Completely reimplement `_print_Pow` from sympy, since it doesn't
+        # correctly handle precision
+        if "Pow" in self.known_functions:
+            return self._print_Function(expr)
+        PREC = precedence(expr)
+        suffix = 'f' if self.single_prec(expr) else ''
+        if equal_valued(expr.exp, -1):
+            return f'{self._print_Float(Float(1.0))}/' + \
+                f'{self.parenthesize(expr.base, PREC)}'
+        elif equal_valued(expr.exp, 0.5):
+            return f'{self._ns}sqrt{suffix}({self._print(expr.base)})'
```

Comment:
What's _ns again?

Reply:
The name space, this is just lifted straight from the sympy code

Reply:
I don't think we handle namespaces here through the printer but directly at method level if needed so I would remove it for now. Only std one might be needed but it's default so not really needed